Adding Copy as Latex plugin

mo-seph opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

Link to my plugin: https://github.com/mo-seph/obsidian-copy-as-latex

## Release Checklist

- [X] Only tested on macOS
- [X] My GitHub release contains all required files
  - [X] `main.js`
  - [X] `manifest.json`
  - [x] `styles.css` _(optional)_
- [X] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [X] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [X] README clearly describes the plugins purpose and provides clear usage instructions.
- [X] I have added a license in the LICENSE file.

lishid commented:
https://github.com/mo-seph/obsidian-copy-as-latex/blob/c9685a6271e57bdcd3362f6acb64535a0f446a9b/src/main.ts#L25
See [this tip](https://github.com/obsidianmd/obsidian-releases/blob/master/plugin-review.md#class-names)

https://github.com/mo-seph/obsidian-copy-as-latex/blob/c9685a6271e57bdcd3362f6acb64535a0f446a9b/src/main.ts#L13
See [this tip](https://github.com/obsidianmd/obsidian-releases/blob/master/plugin-review.md#nodejs-and-electron-api)

https://github.com/mo-seph/obsidian-copy-as-latex/blob/c9685a6271e57bdcd3362f6acb64535a0f446a9b/src/main.ts#L69
https://github.com/mo-seph/obsidian-copy-as-latex/blob/c9685a6271e57bdcd3362f6acb64535a0f446a9b/src/main.ts#L58
I think a fair amount of this setup/initialization can be done when your plugin loads so that you don't have to re-create remark and your conversion functions every time.

mo-seph commented:
> https://github.com/mo-seph/obsidian-copy-as-latex/blob/c9685a6271e57bdcd3362f6acb64535a0f446a9b/src/main.ts#L25 See [this tip](https://github.com/obsidianmd/obsidian-releases/blob/master/plugin-review.md#class-names)
Fixed!


> https://github.com/mo-seph/obsidian-copy-as-latex/blob/c9685a6271e57bdcd3362f6acb64535a0f446a9b/src/main.ts#L13 See [this tip](https://github.com/obsidianmd/obsidian-releases/blob/master/plugin-review.md#nodejs-and-electron-api)

That's really good info, thanks - it took a while to figure out how to do clipboard access.

> https://github.com/mo-seph/obsidian-copy-as-latex/blob/c9685a6271e57bdcd3362f6acb64535a0f446a9b/src/main.ts#L69 https://github.com/mo-seph/obsidian-copy-as-latex/blob/c9685a6271e57bdcd3362f6acb64535a0f446a9b/src/main.ts#L58 I think a fair amount of this setup/initialization can be done when your plugin loads so that you don't have to re-create remark and your conversion functions every time.

I've moved these out of the class, and only make them once now.

Thanks for the tips!

lishid commented:
Seems like you had another plugin tacked in to the PR - please send a separate PR for that one 👍
